### PR TITLE
rbd/bench: outputs bytes/s format dynamically

### PR DESCRIPTION
**Pull request by zy751713126**
Fixes: https://tracker.ceph.com/issues/40256
Signed-off-by: Zheng Yin <zhengyin@cmss.chinamobile.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by zy751713126** [comment]
output ：

[root@cephdev build]# ./bin/rbd bench volumes/test1 --io-type read --io-total 100M
bench  type read io_size 4096 io_threads 16 bytes 104857600 pattern sequential
  SEC       OPS   OPS/SEC   BYTES
    1      2512   2530.53   9.9 MiB/s
    2      5408   2707.94    11 MiB/s
    3      8320   2777.74    11 MiB/s
    4     11232   2809.19    11 MiB/s
    5     13984   2798.88    11 MiB/s
    6     17664   3030.40    12 MiB/s
    7     22304   3380.55    13 MiB/s
elapsed:     7  ops:    25600  ops/sec:  3304.08  bytes/sec: 13533505.42

File: src/tools/rbd/action/Bench.cc [hunk 1]
```diff
@@ -308,7 +308,7 @@ int do_bench(librbd::Image& image, io_type_t io_type,
     op_flags = LIBRADOS_OP_FLAG_FADVISE_SEQUENTIAL;
   }
 
-  printf("  SEC       OPS   OPS/SEC   BYTES/SEC\n");
+  printf("  SEC       OPS   OPS/SEC   BYTES\n");
```
Comment (dillaman) [inline src/tools/rbd/action/Bench.cc]: This is BYTES/SEC -- why change it to the wrong units?

File: src/tools/rbd/action/Bench.cc [hunk 2]
```diff
@@ -383,11 +383,13 @@ int do_bench(librbd::Image& image, io_type_t io_type,
...
       double time_sum = boost::accumulators::rolling_sum(time_acc);
-      printf("%5d  %8d  %8.2lf  %8.2lf\n",
+      printf("%5d  %8d  %8.2lf",
              (int)elapsed.count(),
              (int)(ios - io_threads),
-             boost::accumulators::rolling_sum(ios_acc) / time_sum,
-             boost::accumulators::rolling_sum(off_acc) / time_sum);
+             boost::accumulators::rolling_sum(ios_acc) / time_sum);
+      std::cout.width(10);
+      std::cout << byte_u_t(boost::accumulators::rolling_sum(off_acc) / time_sum) << "/s"
```
Comment (dillaman) [inline src/tools/rbd/action/Bench.cc]: Avoid mixing and matching C and C++ styles -- if you want to help change this over to C++, can you do the whole output line?

**Comment by nikosmeds** [comment]
Any chance this can be optional or a flag could force all results to bytes?

I have a simple script which parses the `rbd bench` results and formats the results for Prometheus ingestion. It would be nice to have the option to keep output consistent between results (either bytes/s, or MiB/s).

**Pull request merged**